### PR TITLE
Change sample runs in ex6p [ex6p-runs2-dev]

### PR DESCRIPTION
**Pull request by jakubcerveny**
This is a follow-up PR to #2105 that actually changes the runs (prefer NC refinement, improve SFC ordering).

(#2105 needs to be merged first)

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2129](https://github.com/mfem/mfem/pull/2129) | @jakubcerveny | @tzanio | @tzanio + @v-dobrev | 04/09/21 | 04/14/21 | 04/18/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by tzanio — APPROVED** [review]

File: examples/ex6p.cpp [hunk 1]
```diff
@@ -2,19 +2,19 @@
...
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 1
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 2
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 2 -h1
 //               mpirun -np 4 ex6p -m ../data/square-disc-nurbs.mesh -o 2
-//               mpirun -np 4 ex6p -m ../data/star.mesh -o 3
-//               mpirun -np 4 ex6p -m ../data/escher.mesh -o 2 -cs
-//               mpirun -np 4 ex6p -m ../data/escher.mesh -o 2
 //               mpirun -np 4 ex6p -m ../data/fichera.mesh -o 2
+//               mpirun -np 4 ex6p -m ../data/escher.mesh -rm 2 -o 2
+//               mpirun -np 4 ex6p -m ../data/escher.mesh -o 2 -cs
```
Comment (v-dobrev) [inline examples/ex6p.cpp:12]: This diff shows re-ordering of the sample runs similar to what was done in #2105. Any idea why it is showing like this?

When I diff this locally with
```console
git diff master...ex6p-runs2-dev
```
I get:
```diff
diff --git a/examples/ex6p.cpp b/examples/ex6p.cpp
index 17a694a94..14f156fd2 100644
--- a/examples/ex6p.cpp
+++ b/examples/ex6p.cpp
@@ -2,19 +2,19 @@
 //
 // Compile with: make ex6p
 //
-// Sample runs:  mpirun -np 4 ex6p -m ../data/star.mesh -o 3
-//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -o 1 -cs
-//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -o 2 -cs
-//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -o 2
+// Sample runs:  mpirun -np 4 ex6p -m ../data/star-hilbert.mesh -o 2
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 1
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 2
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 2 -h1
 //               mpirun -np 4 ex6p -m ../data/square-disc-nurbs.mesh -o 2
 //               mpirun -np 4 ex6p -m ../data/fichera.mesh -o 2
-//               mpirun -np 4 ex6p -m ../data/escher.mesh -o 2
+//               mpirun -np 4 ex6p -m ../data/escher.mesh -rm 2 -o 2
 //               mpirun -np 4 ex6p -m ../data/escher.mesh -o 2 -cs
 //               mpirun -np 4 ex6p -m ../data/disc-nurbs.mesh -o 2
 //               mpirun -np 4 ex6p -m ../data/ball-nurbs.mesh
 //               mpirun -np 4 ex6p -m ../data/pipe-nurbs.mesh
 //               mpirun -np 4 ex6p -m ../data/star-surf.mesh -o 2
-//               mpirun -np 4 ex6p -m ../data/square-disc-surf.mesh -o 2 -cs
+//               mpirun -np 4 ex6p -m ../data/square-disc-surf.mesh -rm 2 -o 2
 //               mpirun -np 4 ex6p -m ../data/inline-segment.mesh -o 1 -md 200
 //               mpirun -np 4 ex6p -m ../data/amr-quad.mesh
 //               mpirun -np 4 ex6p --restart
```

**Comment by v-dobrev** [comment]
What is difference between using `-m ../data/star-hilbert.mesh` and `-m ../data/star.mesh -rm 1`?

File: examples/ex6p.cpp (hunk 1 above)
Comment (jakubcerveny) [reply examples/ex6p.cpp]: I tried merging master into this branch, seems fixed now.

**Comment by jakubcerveny** [comment]
`star-hilbert.mesh` is manually ordered and oriented to get a nice continuous SFC (I wasn't able to get this result automatically with Gecko). The option `-rm 1` is just a quick spatial sort, better than nothing but by far not optimal.

File: examples/ex6p.cpp [hunk 2]
```diff
@@ -2,19 +2,19 @@
...
 // Compile with: make ex6p
 //
-// Sample runs:  mpirun -np 4 ex6p -m ../data/star.mesh -o 3
-//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -o 1 -cs
-//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -o 2 -cs
-//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -o 2
+// Sample runs:  mpirun -np 4 ex6p -m ../data/star-hilbert.mesh -o 2
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 1
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 2
+//               mpirun -np 4 ex6p -m ../data/square-disc.mesh -rm 1 -o 2 -h1
```
Comment (v-dobrev) [inline examples/ex6p.cpp:8]: I think we should keep at least one `-cs` sample run -- we want to test that part of the code too.

Comment (jakubcerveny) [reply examples/ex6p.cpp:8]: There's `../data/escher.mesh -o 2 -cs`, do you want to add more?

Comment (v-dobrev) [reply examples/ex6p.cpp:8]: That's 3D (tets). I was referring to the above 2D (triangles) sample runs.

Comment (jakubcerveny) [reply examples/ex6p.cpp:8]: Ah, OK. I guess the third run could be `-rm 1 -o 2 -cs` (right now it's too similar to the second run anyway).

**Comment by v-dobrev** [comment]
LGTM

**Review by v-dobrev — APPROVED** [review]

**Comment by jakubcerveny** [comment]
Sorry, one last change, the `-rm` option is not needed for the `-cs` run...

**Comment by v-dobrev** [comment]
LGTM

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**